### PR TITLE
Editon 2021 enables precise capture

### PR DESCRIPTION
**Pull request by arora-aman**
r? @nikomatsakis

### CODE REVIEW

File: compiler/rustc_mir_build/src/build/expr/as_place.rs [hunk 1]
```diff
@@ -217,6 +217,10 @@ fn to_upvars_resolved_place_builder<'a, 'tcx>(
                 ty::ClosureKind::FnOnce => {}
             }
 
+            // We won't be building MIR if the closure wasn't local
```
Comment (klensy) [inline compiler/rustc_mir_build/src/build/expr/as_place.rs:220]: Can call to `tcx.hir().local_def_id_to_hir_id(closure_def_id.expect_local())` be deduplicated?

Comment (arora-aman) [reply compiler/rustc_mir_build/src/build/expr/as_place.rs:220]: Right didn't notice there was one burried there. Thanks for catching that

Comment (arora-aman) [reply compiler/rustc_mir_build/src/build/expr/as_place.rs:220]: fixed

**Review by nikomatsakis — APPROVED** [review]

**Pull request merged**